test/Microsoft.DotNet.Darc.VirtualMonoRepo.E2E.Tests/VmrRepositoryCloningTest.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.DotNet.DarcLib.VirtualMonoRepo;
+using Microsoft.Extensions.DependencyInjection;
+using NUnit.Framework;
+
+namespace Microsoft.DotNet.Darc.VirtualMonoRepo.E2E.Tests;
+internal class VmrRepositoryCloningTest : VmrTestsBase
+{
+    [Test]
+    [TestCase(true)]
+    [TestCase(false)]
+    public async Task RepositoryBranchShouldResetToRemote(bool resetToRemote)
+    {
+        var branchName = "main";
+        ILocalGitRepo repo = null!;
```

Comment:
Can we name it better? For instance `clone` or `clonedRepo` so that we know which one is the downstream one?